pkgs/by-name/to/touche/package.nix

```diff
@@ -0,0 +1,85 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchNpmDeps,
+
+  meson,
+  ninja,
+  pkg-config,
+  nodejs,
+  npmHooks,
+  desktop-file-utils,
+  appstream-glib,
+  gjs,
+  gobject-introspection,
+  glib,
+  gtk3,
+  python3,
+  wrapGAppsHook4,
+
+  libadwaita,
+  xorg,
+}:
+let
+  fqpn = "com.github.joseexposito.touche";
+in
+stdenv.mkDerivation (finalAttrs: {
```

Comment:
I don't know much about packaging JS-based projects, but is there any reason to use `mkDerivation` instead of something language-specific like `buildNpmPackage`?

Reply:
buildNpmPackage assumes a lot of things (such as packages will be built via npm) that are just not true here with a GNOME-style Meson setup. Other GNOME/GTK apps in Nixpkgs, regardless of language (C++/Rust/Python/...) follow similar setups.